components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.hpp

```diff
@@ -882,32 +919,31 @@ static void rrtmgp_lw(
   // Weights and angle secants for first order (k=1) Gaussian quadrature.
   //   Values from Table 2, Clough et al, 1992, doi:10.1029/92JD01419
   //   after Abramowitz & Stegun 1972, page 921
-  int constexpr max_gauss_pts = 4;
-  hview_t<RealT**> gauss_Ds_host ("gauss_Ds" ,max_gauss_pts,max_gauss_pts);
-  gauss_Ds_host(0,0) = 1.66      ; gauss_Ds_host(1,0) =         0.; gauss_Ds_host(2,0) =         0.; gauss_Ds_host(3,0) =         0.;
-  gauss_Ds_host(0,1) = 1.18350343; gauss_Ds_host(1,1) = 2.81649655; gauss_Ds_host(2,1) =         0.; gauss_Ds_host(3,1) =         0.;
-  gauss_Ds_host(0,2) = 1.09719858; gauss_Ds_host(1,2) = 1.69338507; gauss_Ds_host(2,2) = 4.70941630; gauss_Ds_host(3,2) =         0.;
-  gauss_Ds_host(0,3) = 1.06056257; gauss_Ds_host(1,3) = 1.38282560; gauss_Ds_host(2,3) = 2.40148179; gauss_Ds_host(3,3) = 7.15513024;
-
-  hview_t<RealT**> gauss_wts_host("gauss_wts",max_gauss_pts,max_gauss_pts);
-  gauss_wts_host(0,0) = 0.5         ; gauss_wts_host(1,0) = 0.          ; gauss_wts_host(2,0) = 0.          ; gauss_wts_host(3,0) = 0.          ;
-  gauss_wts_host(0,1) = 0.3180413817; gauss_wts_host(1,1) = 0.1819586183; gauss_wts_host(2,1) = 0.          ; gauss_wts_host(3,1) = 0.          ;
-  gauss_wts_host(0,2) = 0.2009319137; gauss_wts_host(1,2) = 0.2292411064; gauss_wts_host(2,2) = 0.0698269799; gauss_wts_host(3,2) = 0.          ;
-  gauss_wts_host(0,3) = 0.1355069134; gauss_wts_host(1,3) = 0.2034645680; gauss_wts_host(2,3) = 0.1298475476; gauss_wts_host(3,3) = 0.0311809710;
-
-  view_t<RealT**> gauss_Ds ("gauss_Ds" ,max_gauss_pts,max_gauss_pts);
-  view_t<RealT**> gauss_wts("gauss_wts",max_gauss_pts,max_gauss_pts);
+  RealT gauss_Ds_host_raw[max_gauss_pts][max_gauss_pts] = {
+    {1.66, 1.18350343, 1.09719858, 1.06056257},
+    {0., 2.81649655, 1.69338507, 1.38282560},
+    {0., 0., 4.70941630, 2.40148179},
+    {0., 0., 0., 7.15513024}
+  };
+  hview_t<RealT**> gauss_Ds_host (&gauss_Ds_host_raw[0][0], max_gauss_pts, max_gauss_pts);
+
+  RealT gauss_wts_host_raw[max_gauss_pts][max_gauss_pts] = {
+    {0.5, 0.3180413817, 0.2009319137, 0.1355069134},
+    {0., 0.1819586183, 0.2292411064, 0.2034645680},
+    {0., 0., 0.0698269799, 0.1298475476},
+    {0., 0., 0., 0.0311809710}
+  };
+
+  hview_t<RealT**> gauss_wts_host(&gauss_wts_host_raw[0][0],max_gauss_pts,max_gauss_pts);
```

Comment:
Are we copying the same gauss quadrature info to device over and over at every time step? Can we do this at init as well, and then pass around the pre-filled views at runtime? It seems silly to set them up every time...

Reply:
We could do this at init but I don't think this is very expensive at all.

Reply:
I agree that it's not very expensive, but so is a single device allocation (in the grand scheme of things). It's just something we can remove, and these small opt all pile up. But yeah, no need to do it here.

Reply:
@bartgol , i don't think there's any allocation happening here. The C arrays will be on the stack and the views just point to that memory.

Reply:
There's no alloc, correct, but there is are two deep_copy, which involve a small kernel launch. Nothing big, but also pointless.